skimage/measure/fit.py

```diff
@@ -282,6 +282,15 @@ def estimate(self, data):
         # to prevent integer overflow, cast data to float, if it isn't already
         float_type = np.promote_types(data.dtype, np.float32)
         data = data.astype(float_type, copy=False)
+        # normalize value range to avoid misfitting due to numeric errors if
+        # the relative distanceses are small compared to absolute distances
+        origin = data.mean(axis=0)
+        data = data - origin
+        scale = data.std()
+        if scale < np.finfo(float_type).tiny:
+            warn("Cannot resolve circle in data.")
```

Comment:
I think this message could be more specific. Perhaps "standard deviation of data is too small to estimate circle with meaningfull precision"?